File: wagerproof-ios-native/Wagerproof/Features/MLB/Components/MatchupPropsDetailSheet.swift
import SwiftUI
import WagerproofDesign
import WagerproofModels

/// Full team-grouped player-prop list for one matchup — the expand target of
/// the game sheet's `MLBMatchupPropsWidget` digest. The list body is the old
/// widget body relocated verbatim (away group / home group, 16pt logo headers
/// + counts, divider rows); reusable so Search's props destination can host it
/// inside its own NavigationStack too.
struct MatchupPropsListBody: View {
    let matchup: MLBPropMatchup
    /// When the host's push destination declares a zoom transition, rows
    /// register as matched sources here; nil hosts get plain pushes.
    var zoomNamespace: Namespace.ID? = nil
    let onSelect: (PlayerPropSelection) -> Void

    private var items: [PlayerPropFeedItem] {
        PlayerPropFeed.items(from: [matchup])
    }
    private var awayItems: [PlayerPropFeedItem] {
        items.filter { $0.selection.teamAbbr == matchup.awayAbbr }
    }
    private var homeItems: [PlayerPropFeedItem] {
        items.filter { $0.selection.teamAbbr == matchup.homeAbbr }
    }
    // Extra (bench/pinch-hit) batters get teamAbbr "" from PlayerPropFeed, so
    // they'd vanish from both team groups while the widget footer's
    // totalProps count still advertises them — render them in a third group.
    private var extraItems: [PlayerPropFeedItem] {
        items.filter {
            $0.selection.teamAbbr != matchup.awayAbbr && $0.selection.teamAbbr != matchup.homeAbbr
        }
    }

    var body: some View {
        VStack(spacing: 0) {
            teamGroup(abbr: matchup.awayAbbr, logo: matchup.awayLogoUrl, items: awayItems)
            teamGroup(abbr: matchup.homeAbbr, logo: matchup.homeLogoUrl, items: homeItems)
            teamGroup(abbr: "More props", logo: nil, items: extraItems)
        }
    }

    @ViewBuilder
    private func teamGroup(abbr: String, logo: String?, items: [PlayerPropFeedItem]) -> some View {
        if !items.isEmpty {
            VStack(spacing: 0) {
                HStack(spacing: 6) {
                    if let logo, let url = URL(string: logo) {
                        AsyncImage(url: url) { phase in
                            if case .success(let img) = phase { img.resizable().scaledToFit() } else { Color.clear }
                        }
                        .frame(width: 16, height: 16)
                    }
                    Text(abbr.uppercased())
                        .font(.system(size: 11, weight: .bold))
                        .tracking(0.6)
                        .foregroundStyle(Color.appTextSecondary)
                    Spacer()
                    Text("\(items.count)")
                        .font(.system(size: 11, weight: .semibold))
                        .foregroundStyle(Color.appTextMuted)
                }
                .padding(.top, 10)
                .padding(.bottom, 6)

                ForEach(Array(items.enumerated()), id: \.element.id) { idx, item in
                    if idx > 0 {
                        Divider().background(Color.appBorder.opacity(0.4))
                    }
                    if let ns = zoomNamespace {
                        MatchupPropListRow(item: item, onSelect: onSelect)
                            .matchedTransitionSource(id: item.selection.transitionID, in: ns)
                    } else {
                        MatchupPropListRow(item: item, onSelect: onSelect)
                    }
                }
            }
        }
    }
}

/// Modal expand target presented from the MLB game sheet. Owns its own
/// NavigationStack + zoom namespace so player rows can push the full
/// `PlayerPropDetailView` inside the sheet.
struct MatchupPropsDetailSheet: View {
    let matchup: MLBPropMatchup

    @State private var selectedProp: PlayerPropSelection?
    @Namespace private var propNS

    var body: some View {
        NavigationStack {
            ScrollView {
                VStack(alignment: .leading, spacing: 8) {
                    header
                    MatchupPropsListBody(matchup: matchup, zoomNamespace: propNS) { selectedProp = $0 }
                        .padding(.horizontal, 16)
                }
                .padding(.bottom, 32)
            }
            .background(Color.appSurface)
            .navigationDestination(item: $selectedProp) { selection in
                PlayerPropDetailView(selection: selection)
                    .navigationTransition(.zoom(sourceID: selection.transitionID, in: propNS))
            }
        }
        .presentationDetents([.large])
        .presentationDragIndicator(.visible)
    }

    private var header: some View {
        VStack(alignment: .leading, spacing: 2) {
            Text("\(matchup.awayAbbr) @ \(matchup.homeAbbr) · Player Props")
                .font(.system(size: 18, weight: .bold))
                .foregroundStyle(Color.appTextPrimary)
            Text("\(MLBFormatting.dateLabel(matchup.officialDate)) · \(MLBFormatting.gameTime(matchup.gameTimeEt))")
                .font(.system(size: 12, weight: .semibold))
                .foregroundStyle(Color.appTextSecondary)
        }
        .padding(.horizontal, 16)
        .padding(.top, 20)
    }
}

/// Compact tappable player row — headshot, name, role/opponent, best-market
/// line, and L10 hit-rate. Relocated from the old widget body; the host body
/// registers each row as a zoom source when it has a namespace.
private struct MatchupPropListRow: View {
    let item: PlayerPropFeedItem
    let onSelect: (PlayerPropSelection) -> Void

    private var sel: PlayerPropSelection { item.selection }
    private var computed: MLBPropComputedAtLine { item.headline.computed }
    private var row: MLBPlayerPropRow { item.headline.row }

    private var roleLabel: String {
        if let pos = sel.position, !pos.isEmpty { return pos }
        return sel.isPitcher ? "SP" : ""
    }

    private var hitColor: Color {
        guard let pct = computed.l10.pct else { return Color.appTextMuted }
        if pct >= 70 { return Color.appPrimary }
        if pct >= 55 { return Color(hex: 0xEAB308) }
        return Color.appTextSecondary
    }

    var body: some View {
        Button {
            onSelect(sel)
        } label: {
            HStack(spacing: 10) {
                PlayerHeadshot(playerId: sel.playerId, size: 32)
                    .frame(width: 32, height: 32)
                VStack(alignment: .leading, spacing: 2) {
                    Text(sel.playerName)
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundStyle(Color.appTextPrimary)
                        .lineLimit(1)
                        .minimumScaleFactor(0.8)
                    Text(subtitle)
                        .font(.system(size: 11, weight: .medium))
                        .foregroundStyle(Color.appTextSecondary)
                        .lineLimit(1)
                }
                Spacer(minLength: 8)
                VStack(alignment: .trailing, spacing: 2) {
                    Text("\(MLBPlayerProps.marketLabel(row.market)) \(MLBPlayerProps.formatLine(computed.line))")
                        .font(.system(size: 12, weight: .semibold))
                        .foregroundStyle(Color.appTextPrimary)
                        .lineLimit(1)
                    Text("L10 \(computed.l10.pctLabel)")
                        .font(.system(size: 11, weight: .bold))
                        .foregroundStyle(hitColor)
                }
                Image(systemName: "chevron.right")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(Color.appTextMuted)
            }
            .padding(.vertical, 9)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .sensoryFeedback(.impact(weight: .light), trigger: sel.id)
    }

    private var subtitle: String {
        let opp = sel.opponentAbbr.isEmpty ? "" : "vs \(sel.opponentAbbr)"
        if roleLabel.isEmpty { return opp }
        return opp.isEmpty ? roleLabel : "\(roleLabel) · \(opp)"
    }
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/mlb/props/MatchupPropsDetailSheet.kt
package com.wagerproof.app.features.mlb.props

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.wagerproof.app.features.props.PlayerPropFeed
import com.wagerproof.app.features.props.PlayerPropFeedItem
import com.wagerproof.app.features.props.PlayerPropSelection
import com.wagerproof.app.features.props.components.PlayerHeadshot
import com.wagerproof.app.features.shared.RemoteImage
import com.wagerproof.app.features.shared.hexColor
import com.wagerproof.core.design.icons.AppIcon
import com.wagerproof.core.design.tokens.AppColors
import com.wagerproof.core.models.MLBPlayerProps
import com.wagerproof.core.models.MLBPropMatchup
import com.wagerproof.app.features.mlb.MLBFormatting

/**
 * Full team-grouped player-prop list — port of iOS `MatchupPropsDetailSheet` /
 * `MatchupPropsListBody`. The expand target of [MLBMatchupPropsWidget]; rows
 * push the player's prop detail via [onSelect]. Rendered as a full screen.
 */
@Composable
fun MatchupPropsDetailSheet(
    matchup: MLBPropMatchup,
    onSelect: (PlayerPropSelection) -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier
            .fillMaxSize()
            .background(AppColors.appSurface)
            .verticalScroll(rememberScrollState())
            .padding(bottom = 32.dp),
    ) {
        Column(Modifier.padding(horizontal = 16.dp).padding(top = 20.dp), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(
                "${matchup.awayAbbr} @ ${matchup.homeAbbr} · Player Props",
                fontSize = 18.sp, fontWeight = FontWeight.Bold, color = AppColors.appTextPrimary,
            )
            Text(
                "${MLBFormatting.dateLabel(matchup.officialDate)} · ${MLBFormatting.gameTime(matchup.gameTimeEt)}",
                fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextSecondary,
            )
        }
        Spacer(Modifier.height(8.dp))
        MatchupPropsListBody(matchup, onSelect, Modifier.padding(horizontal = 16.dp))
    }
}

@Composable
fun MatchupPropsListBody(
    matchup: MLBPropMatchup,
    onSelect: (PlayerPropSelection) -> Unit,
    modifier: Modifier = Modifier,
) {
    val items = PlayerPropFeed.items(listOf(matchup))
    val awayItems = items.filter { it.selection.teamAbbr == matchup.awayAbbr }
    val homeItems = items.filter { it.selection.teamAbbr == matchup.homeAbbr }
    // Extra (bench/pinch-hit) batters get teamAbbr "" — render as a third group.
    val extraItems = items.filter { it.selection.teamAbbr != matchup.awayAbbr && it.selection.teamAbbr != matchup.homeAbbr }

    Column(modifier.fillMaxWidth()) {
        TeamGroup(matchup.awayAbbr, matchup.awayLogoUrl, awayItems, onSelect)
        TeamGroup(matchup.homeAbbr, matchup.homeLogoUrl, homeItems, onSelect)
        TeamGroup("More props", null, extraItems, onSelect)
    }
}

@Composable
private fun TeamGroup(abbr: String, logo: String?, items: List<PlayerPropFeedItem>, onSelect: (PlayerPropSelection) -> Unit) {
    if (items.isEmpty()) return
    Column(Modifier.fillMaxWidth()) {
        Row(Modifier.fillMaxWidth().padding(top = 10.dp, bottom = 6.dp), verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            if (logo != null) {
                RemoteImage(logo, abbr, Modifier.size(16.dp))
            }
            Text(abbr.uppercase(), fontSize = 11.sp, fontWeight = FontWeight.Bold, letterSpacing = 0.6.sp, color = AppColors.appTextSecondary)
            Box(Modifier.weight(1f))
            Text("${items.size}", fontSize = 11.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextMuted)
        }
        items.forEachIndexed { idx, item ->
            if (idx > 0) Box(Modifier.fillMaxWidth().height(0.5.dp).background(AppColors.appBorder.copy(alpha = 0.4f)))
            MatchupPropListRow(item, onSelect)
        }
    }
}

@Composable
private fun MatchupPropListRow(item: PlayerPropFeedItem, onSelect: (PlayerPropSelection) -> Unit) {
    val sel = item.selection
    val computed = item.headline.computed
    val pct = computed.l10.pct
    val hitColor: Color = when {
        pct == null -> AppColors.appTextMuted
        pct >= 70 -> AppColors.appPrimary
        pct >= 55 -> hexColor(0xEAB308L)
        else -> AppColors.appTextSecondary
    }
    val roleLabel = sel.position?.takeIf { it.isNotEmpty() } ?: if (sel.isPitcher) "SP" else ""
    val opp = if (sel.opponentAbbr.isEmpty()) "" else "vs ${sel.opponentAbbr}"
    val subtitle = when {
        roleLabel.isEmpty() -> opp
        opp.isEmpty() -> roleLabel
        else -> "$roleLabel · $opp"
    }
    Row(
        Modifier.fillMaxWidth().clickable { onSelect(sel) }.padding(vertical = 9.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(10.dp),
    ) {
        PlayerHeadshot(playerId = sel.playerId, size = 32.dp)
        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(sel.playerName, fontSize = 14.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextPrimary, maxLines = 1)
            Text(subtitle, fontSize = 11.sp, fontWeight = FontWeight.Medium, color = AppColors.appTextSecondary, maxLines = 1)
        }
        Spacer(Modifier.width(8.dp))
        Column(horizontalAlignment = Alignment.End, verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(
                "${MLBPlayerProps.marketLabel(item.headline.row.market)} ${MLBPlayerProps.formatLine(computed.line)}",
                fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = AppColors.appTextPrimary, maxLines = 1,
            )
            Text("L10 ${computed.l10.pctLabel}", fontSize = 11.sp, fontWeight = FontWeight.Bold, color = hitColor)
        }
        Icon(AppIcon.CHEVRON_RIGHT.imageVector, null, tint = AppColors.appTextMuted, modifier = Modifier.size(12.dp))
    }
}
